pkg/controller/certificates/sync.go

```diff
@@ -114,20 +124,51 @@ func (c *Controller) Sync(ctx context.Context, crt *v1alpha1.Certificate) (err e
 		return err
 	}
 
+	// update certificate expiry metric
+	defer c.updateCertificateExpiryMetric(crt)
+
 	// as there is an existing certificate, or we may create one below, we will
 	// run scheduleRenewal to schedule a renewal if required at the end of
 	// execution.
 	defer c.scheduleRenewal(crt)
 
 	crtCopy := crt.DeepCopy()
 
+	// get issuer details for metrics
+	issuerName := issuerObj.GetObjectMeta().Name
+	var issuerKind string
+	switch crt.Spec.IssuerRef.Kind {
+	case "", v1alpha1.IssuerKind:
+		issuerKind = v1alpha1.IssuerKind
+	case v1alpha1.ClusterIssuerKind:
+		issuerKind = v1alpha1.ClusterIssuerKind
+	default:
+		return fmt.Errorf(`invalid value %q for certificate issuer kind. Must be empty, %q or %q`, crt.Spec.IssuerRef.Kind, v1alpha1.IssuerKind, v1alpha1.ClusterIssuerKind)
+	}
```

Comment:
Can we remove this block altogether - instead, I'd like to see the `metrics` package do extra work in order to obtain this information. Metrics should never cause errors to be returned as part of a sync - they should be provided on a best-effort basis.